client/layout/masterbar/minimal.jsx

```diff
@@ -3,24 +3,35 @@
 /**
  * External dependencies
  */
-import React from 'react';
+import React from 'react/addons';
 import Masterbar from './masterbar';
 
 /**
  * Internal dependencies
  */
 import Item from './item';
+import config from 'config';
 
-const MasterbarMinimal = ( { url } ) => (
-	<Masterbar>
-		<Item url={ url } icon="my-sites" className="masterbar__item-logo">
-			WordPress<span className="tld">.com</span>
-		</Item>
-	</Masterbar>
-);
+export default React.createClass( {
+	propTypes: {
+		url: React.PropTypes.string.isRequired
+	},
 
-MasterbarMinimal.propTypes = {
-	url: React.PropTypes.string.isRequired
-};
-
-export default MasterbarMinimal;
+	render() {
+		return(
+			<Masterbar>
+				<Item url={ this.props.url } icon="my-sites" className="masterbar__item-logo is-loggedout">
+					WordPress<span className="tld">.com</span>
+				</Item>
+				<div className="masterbar__login-links">
```

Comment:
As you pointed out, any elements added here will be displayed on checkout, which is currently only displayed when the user is logged in.

The best option I can think of here is to extend `MasterbarMinimal` to render its children here, and create a `MasterbarLoggedOut` wrapper that provides the child elements (i.e. the sign up/log in links).